docs/openapi-docs/src/test/scala-2/sttp/tapir/docs/openapi/VerifyYamlEnumeratumTest.scala

```diff
@@ -31,7 +31,74 @@ class VerifyYamlEnumeratumTest extends AnyFunSuite with Matchers {
     val expectedYaml = load("validator/expected_valid_enumeratum_with_metadata.yml")
 
     val actualYaml =
-      OpenAPIDocsInterpreter().toOpenAPI(endpoint.in("numbers").in(jsonBody[Enumeratum.NumberWithMsg]), Info("Numbers", "1.0")).toYaml
+      OpenAPIDocsInterpreter()
+        .toOpenAPI(endpoint.in("numbers").in(jsonBody[Enumeratum.NumberWithMsg]), Info("Numbers", "1.0"))
+        .toYaml
+
+    noIndentation(actualYaml) shouldBe expectedYaml
+  }
+
+  // #1800
+  test("add enum default") {
+    import sttp.tapir.codec.enumeratum._
+    import sttp.tapir.docs.openapi.VerifyYamlEnumeratumTest.Enumeratum.FruitType._
+
+    val expectedYaml = load("enum/expected_enumeratum_enum_default.yml")
+    val ep = endpoint
+      .in(query[Enumeratum.FruitType]("type").example(APPLE).default(PEAR))
+      .out(jsonBody[Enumeratum.FruitWithEnum])
+
+    val actualYaml =
+      OpenAPIDocsInterpreter().toOpenAPI(ep, Info("Fruits", "1.0")).toYaml
+
+    noIndentation(actualYaml) shouldBe expectedYaml
+  }
+
+  // #1800
+  test("use enum default if more than one") {
+    import sttp.tapir.codec.enumeratum._
+    import sttp.tapir.docs.openapi.VerifyYamlEnumeratumTest.Enumeratum.FruitType._
+
+    val expectedYaml = load("enum/expected_enumeratum_enum_default_if_more_than_one.yml")
+    val ep1 = endpoint
+      .in("fruit-by-type1").in(query[Enumeratum.FruitType]("type1").default(PEAR))
+      .out(jsonBody[Enumeratum.FruitWithEnum])
+    val ep2 = endpoint
+      .in("fruit-by-type2").in(query[Enumeratum.FruitType]("type2").default(APPLE))
+      .out(jsonBody[Enumeratum.FruitWithEnum])
+
+    val actualYaml =
+      OpenAPIDocsInterpreter().toOpenAPI(List(ep1, ep2), Info("Fruits", "1.0")).toYaml
+
+    noIndentation(actualYaml) shouldBe expectedYaml
+  }
+
+  // #1800
+  test("ignore enum default in request body") {
+    import sttp.tapir.codec.enumeratum._
+
+    val expectedYaml = load("enum/expected_enumeratum_enum_default_in_request_body.yml")
+    val ep = endpoint
+      .post.in(jsonBody[Enumeratum.FruitQuery])
+      .out(jsonBody[Enumeratum.FruitWithEnum])
+
+    val actualYaml =
+      OpenAPIDocsInterpreter().toOpenAPI(ep, Info("Fruits", "1.0")).toYaml
+
+    noIndentation(actualYaml) shouldBe expectedYaml
+  }
+
+  // #1800
+  test("add enum default in request body with given encoded") {
```

Comment:
So it matches with case above - "should add default when encoded value specified"